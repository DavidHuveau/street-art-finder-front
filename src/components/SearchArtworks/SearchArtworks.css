#Srch-wrapper {
  /* border: 1px solid blue; */
}
