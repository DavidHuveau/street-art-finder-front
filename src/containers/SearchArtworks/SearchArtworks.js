import React, { Component } from "react";

class SearchArtworks extends Component {
  render() {
    return <div>SearchArtworks</div>;
  }
}

export default SearchArtworks;
