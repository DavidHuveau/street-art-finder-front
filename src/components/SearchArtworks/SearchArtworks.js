import React, { Component } from "react";
import "./SearchArtworks.css";
import { InputGroup, Button, FormControl, Form } from "react-bootstrap";
import PropTypes from "prop-types";

class SearchArtworks extends Component {
  constructor(props) {
    super(props);

    ["onChangeCriteria", "onSubmit", "onClickClear", "onKeyPress"].forEach(
      fn => (this[fn] = this[fn].bind(this))
    );

    this.state = {
      valueCriteria: "" // only City in first time
    };
  }

  onChangeCriteria({target}) {
    const { value } = target;
    this.setState({
      valueCriteria: value
    });
  }

  onSubmit(event) {
    event.preventDefault();
    const { valueCriteria } = this.state;
    const { onSearch } = this.props;

    if (valueCriteria) onSearch(valueCriteria);
  }

  onClickClear() {
    this.setState({
      valueCriteria: ""
    });
  }

  onKeyPress(event) {
    if (event.key === 'Enter') {
      event.stopPropagation();
      this.onSubmit(event);
    }
  }

  render() {
    const { valueCriteria } = this.state;

    return (
      <Form id="Srch-wrapper" onSubmit={this.onSubmit}>
        <InputGroup>
          <FormControl
            type="text"
            placeholder="City"
            value={valueCriteria}
            onChange={this.onChangeCriteria}
            onKeyPress={this.onKeyPress}
          />
          { valueCriteria.length > 0 && <InputGroup.Append>
              <Button variant="outline-secondary" onClick={this.onClickClear}>
                <i className="fa fa-eraser"></i>
              </Button>
            </InputGroup.Append>
          }
          <InputGroup.Append>
            <Button
              variant="outline-secondary"
              type="submit"
              id="submit-button"
            >
              <i className="fa fa-search"></i>
            </Button>
          </InputGroup.Append>
          <div className="separator">
          </div>
          <a href="https://www.instagram.com/street.art.finder/" target="_blank" style={{ margin:"auto" }}>
            <i className="fab fa-instagram" style={{ fontSize:"17px" }}></i>
          </a>
        </InputGroup>
      </Form>
    );
  }
}

export default SearchArtworks;

SearchArtworks.propTypes = {
  onSearch: PropTypes.func.isRequired
};

SearchArtworks.defaultProps = {
  onSearch: () => {}
};
