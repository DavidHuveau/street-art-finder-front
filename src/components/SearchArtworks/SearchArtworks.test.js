import React from "react";
import { mount } from 'enzyme';
import SearchArtworks from "./SearchArtworks";

describe("Check Components", () => {
  it("Component SearchArtworks", () => {
    const component = mount(<SearchArtworks />);
    component.find("input").simulate('change', {
      target: { value: 'Reims' }
    })

    expect(component.state('valueCriteria')).toEqual('Reims')
    expect(component).toMatchSnapshot();

    component.unmount();
  })
})

