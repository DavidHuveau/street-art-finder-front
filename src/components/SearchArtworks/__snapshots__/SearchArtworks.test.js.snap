// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Components Component SearchArtworks 1`] = `
<SearchArtworks
  onSearch={[Function]}
>
  <div
    className="left-lg right-lg"
    id="Srch-wrapper"
  >
    <InputGroup>
      <div
        className="input-group"
      >
        <InputGroupPrepend>
          <div
            className="input-group-prepend"
          >
            <Button
              active={false}
              disabled={false}
              onClick={[Function]}
              type="submit"
              variant="outline-secondary"
            >
              <button
                className="btn btn-outline-secondary"
                disabled={false}
                onClick={[Function]}
                type="submit"
              >
                Search
              </button>
            </Button>
          </div>
        </InputGroupPrepend>
        <FormControl
          onChange={[Function]}
          placeholder="City"
          type="text"
          value="Reims"
        >
          <input
            className="form-control"
            onChange={[Function]}
            placeholder="City"
            type="text"
            value="Reims"
          />
        </FormControl>
        <InputGroupAppend>
          <div
            className="input-group-append"
          >
            <Button
              active={false}
              disabled={false}
              onClick={[Function]}
              type="button"
              variant="outline-secondary"
            >
              <button
                className="btn btn-outline-secondary"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                X
              </button>
            </Button>
          </div>
        </InputGroupAppend>
      </div>
    </InputGroup>
  </div>
</SearchArtworks>
`;
